lib/charms/tls_certificates_interface/v2/tls_certificates.py

```diff
@@ -1057,6 +1058,47 @@ def remove_certificate(self, certificate: str) -> None:
         for certificate_relation in certificates_relation:
             self._remove_certificate(certificate=certificate, relation_id=certificate_relation.id)
 
+    def get_all_relations_certificates(self) -> Dict[str, Dict[str, str]]:
+        """Returns a dictionary of all issued certificates.
+
+        Certificates are returned per application name and CSR.
+
+        Returns:
+            dict: Certificates per application name.
+        """
+        certificates: Dict[str, Dict[str, str]] = defaultdict(dict)
+        for relation in self.model.relations[self.relationship_name]:
```

Comment:
Iirc, this could raise a KeyError. How about:
```suggestion
        for relation in self.model.relations.get(self.relationship_name, []):
```

Reply:
✅